### PR TITLE
Cherry-pick Babelfish-related commits from BABEL_1_X_DEV__PG_13_6 - part 1

### PR DESCRIPTION
**Pull request by 2jungkook**
Cherry-pick `Initial commit for Babelfish 1.x development (3e4958b416)` and resolve merge conflicts.

Author: Jungkook Lee <jungkook@amazon.com>
 
### Issues Resolved

N/A
 
### Check List

- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the PostgreSQL license, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by 2jungkook** [comment]
Merge-conflict: [pg14-merge-conflict.txt](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/files/8334531/pg14-merge-conflict.txt)
Other than the merge conflict, all changes were already reviewed.

**Comment by 2jungkook** [comment]
In this PR, I resolved merge conflicts on 22 files (attached diff in the previous comment).

@sangilsong, please review the following files:
```
src/backend/commands/functioncmds.c
src/backend/commands/trigger.c
src/backend/executor/nodeModifyTable.c
src/backend/nodes/copyfuncs.c
src/backend/parser/analyze.c
src/backend/parser/gram.y
src/backend/utils/misc/guc.c
src/include/nodes/execnodes.h
src/include/utils/plancache.h
src/backend/postmaster/pgstat.c
```

and @JimFinnerty, please review
```
src/backend/access/common/printtup.c
src/backend/access/transam/xact.c
src/backend/catalog/pg_collation.c
src/backend/libpq/pqcomm.c
src/backend/parser/parse_node.c
src/backend/parser/parse_relation.c
src/backend/tcop/postgres.c
src/fe_utils/psqlscan.l
src/include/catalog/pg_collation.h
src/include/libpq/libpq.h
src/include/nodes/parsenodes.h
src/include/pgstat.h
```

File: src/backend/parser/analyze.c [hunk 1]
```diff
@@ -1236,7 +1301,7 @@ count_rowexpr_columns(ParseState *pstate, Node *expr)
  * see below for the other cases.
  */
 static Query *
-transformSelectStmt(ParseState *pstate, SelectStmt *stmt)
+transformSelectStmt(ParseState *pstate, SelectStmt *stmt) ///
```
Comment (sangilsong) [inline src/backend/parser/analyze.c]: can we avoid unnecessary touch?

File: src/backend/parser/analyze.c [hunk 2]
```diff
@@ -1267,7 +1332,7 @@ transformSelectStmt(ParseState *pstate, SelectStmt *stmt)
 	pstate->p_windowdefs = stmt->windowClause;
 
 	/* process the FROM clause */
-	transformFromClause(pstate, stmt->fromClause);
+	transformFromClause(pstate, stmt->fromClause); ///
```
Comment (sangilsong) [inline src/backend/parser/analyze.c]: nit

File: src/backend/parser/analyze.c [hunk 3]
```diff
@@ -2353,11 +2418,14 @@ transformUpdateStmt(ParseState *pstate, UpdateStmt *stmt)
...
-	qual = transformWhereClause(pstate, stmt->whereClause,
+	/* Check if self-join transformation is needed for update satatement with output clause */
+	if (pre_output_clause_transformation_hook)
+		qual = (*pre_output_clause_transformation_hook) (pstate, stmt, qry->commandType);
+	else
+		qual = transformWhereClause(pstate, stmt->whereClause,
 								EXPR_KIND_WHERE, "WHERE");
 
 	qry->returningList = transformReturningList(pstate, stmt->returningList);
-
```
Comment (sangilsong) [inline src/backend/parser/analyze.c]: nit

File: src/backend/utils/misc/guc.c [hunk 4]
```diff
@@ -584,7 +594,7 @@ int			log_min_duration_statement = -1;
 int			log_parameter_max_length = -1;
 int			log_parameter_max_length_on_error = 0;
 int			log_temp_files = -1;
-double		log_statement_sample_rate = 1.0;
+double      log_statement_sample_rate = 1.0;
```
Comment (sangilsong) [inline src/backend/utils/misc/guc.c]: nit

**Review by sangilsong — DISMISSED** [review]

**Review by JimFinnerty — DISMISSED** [review]

File: src/backend/parser/analyze.c (hunk 1 above)
Comment (2jungkook) [reply src/backend/parser/analyze.c]: done

File: src/backend/parser/analyze.c (hunk 2 above)
Comment (2jungkook) [reply src/backend/parser/analyze.c]: done

File: src/backend/parser/analyze.c (hunk 3 above)
Comment (2jungkook) [reply src/backend/parser/analyze.c]: done

File: src/backend/utils/misc/guc.c (hunk 4 above)
Comment (2jungkook) [reply src/backend/utils/misc/guc.c]: done

**Review by sangilsong — APPROVED** [review]

**Review by JimFinnerty — APPROVED** [review]

**Pull request merged**